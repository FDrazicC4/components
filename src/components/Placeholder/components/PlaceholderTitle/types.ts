import { ReactNode } from 'react';


export interface Props {

    /**
     * Classes to be passed through to the component.
     */
    className?: string,

    /**
     * The title content of the placeholder.
     */
    children: ReactNode,

}
