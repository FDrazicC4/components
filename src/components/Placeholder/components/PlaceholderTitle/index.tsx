import React, { ReactElement } from 'react';
import clsx from 'clsx';
import Typography from '../../../Typography';
import { Props } from './types';


function PlaceholderTitle({
    className,
    children,
}: Props): ReactElement {


    const placeholderClasses = clsx(
        'brew-Placeholder__title',
        className,
    );


    return (
        <Typography className={placeholderClasses} color="tertiary" shouldTruncate variant="h1">
            {children}
        </Typography>
    );


}


export default PlaceholderTitle;
