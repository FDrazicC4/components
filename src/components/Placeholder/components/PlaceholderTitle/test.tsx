import React from 'react';
import renderer from 'react-test-renderer';
import PlaceholderTitle from './index';
import { shallow } from 'enzyme';


describe('PlaceholderTitle', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<PlaceholderTitle>Foo</PlaceholderTitle>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('mounts correctly', () => {
        const placeholderTitle = shallow(<PlaceholderTitle>Foo</PlaceholderTitle>);
        expect(placeholderTitle.hasClass('brew-Placeholder__title')).toBe(true);
    });

    it('correctly adds a className', () => {
        const className = `class-${Math.floor(Math.random() * 1000)}`;
        const placeholderTitle = shallow(<PlaceholderTitle className={className}>Foo</PlaceholderTitle>);
        expect(placeholderTitle.hasClass(className)).toBe(true);
    });

});
