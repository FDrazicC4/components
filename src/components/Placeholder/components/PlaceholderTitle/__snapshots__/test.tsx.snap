// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderTitle renders correctly 1`] = `
<h1
  className="wrapper variant--h1 color--tertiary align--inherit display--initial hasBottomMargin shouldTruncate brew-Placeholder__title"
>
  Foo
</h1>
`;
